File: backend/services/crop_analysis_service.py
from ai.cropAnalysis.model_wrapper import predict_crop_analysis


def analyze_crop_service(image_bytes: bytes, crop_type: str, growth_stage: str) -> dict:
    raw = predict_crop_analysis(image_bytes, crop_type, growth_stage)

    return {
        "crop_type": crop_type,
        "growth_stage": growth_stage,
        "health_score": raw.get("health_score", 0),
        "nutrient_deficiencies": raw.get("nutrient_deficiencies", []),
        "growth_assessment": raw.get("growth_assessment", "No assessment available."),
        "pest_risk": raw.get("pest_risk", "Unknown"),
        "recommendations": raw.get("recommendations", []),
        "confidence_percent": raw.get("confidence_percent", 0),
        "note": raw.get("note", None)
    }

File: frontend/src/services/cropAnalysisService.js
const API_BASE = import.meta.env.VITE_API_URL || 'http://localhost:8000';

export const analyzeCrop = async (formData) => {
  try {
    const response = await fetch(`${API_BASE}/api/crop/analyze`, {
      method: 'POST',
      body: formData,
    });

    if (!response.ok) {
      const details = await response.json().catch(() => null);
      throw new Error(details?.detail || response.statusText);
    }

    return await response.json();
  } catch (error) {
    console.error('Crop analysis error:', error);
    throw error;
  }
};

export const formatCropAnalysisResult = (prediction) => ({
  cropType: prediction.crop_type || 'Unknown',
  growthStage: prediction.growth_stage || 'Unknown',
  healthScore: prediction.health_score ?? 0,
  nutrientDeficiencies: prediction.nutrient_deficiencies || [],
  growthAssessment: prediction.growth_assessment || 'No assessment available.',
  pestRisk: prediction.pest_risk || 'Unknown',
  recommendations: prediction.recommendations || [],
  confidence: prediction.confidence_percent ?? 0,
  note: prediction.note || null,
});
